Library/Homebrew/cmd/log.rb

```diff
@@ -1,15 +1,15 @@
+require "formulary"
+require "formula"
+
 module Homebrew
   def log
     if ARGV.named.empty?
       cd HOMEBREW_REPOSITORY
       exec "git", "log", *ARGV.options_only
     else
-      begin
-        path = ARGV.formulae.first.path
-      rescue FormulaUnavailableError
-        # Maybe the formula was deleted
-        path = Formula.path(ARGV.named.first)
-      end
+      path = Formulary.loader_for(ARGV.named.first).path
```

Comment:
This place is a little different. Original design will have weird response if I do something like this: 
`brew log foo bar` if foo is in the tap and bar doesn't exist. It will assume foo is in the core.